### DISCUSSION TITLE
null safe foreach statement

### DISCUSSION THREAD

**Discussion by foximoxi** [dotnet/csharplang#5144] [Language Ideas] [answered]
I would like to propose C# null-checking foreach statement to avoid extra condition before the loop.

`foreach? (var n in collection) 
{
    //....
}`

**Comment by jnm2** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/discussions/1081

**Reply by TonyValenti** [reply]
I have a "Coalesce" extension method for enumerables that returns enumerable.empty if the source enumerable is null.

**Reply by TonyValenti** [reply]
I have a "Coalesce" extension method for enumerables that returns enumerable.empty if the source enumerable is null.
